### PR TITLE
Modified functionality of DoRestRequest in utils.go to catch the appropriate errors returned by APIC

### PR DESCRIPTION
**Pull request by shrsr**

### CODE REVIEW

**Review by samiib — CHANGES_REQUESTED** [review]

File: internal/provider/utils.go [hunk 1]
```diff
@@ -61,24 +61,34 @@ func DoRestRequestEscapeHtml(ctx context.Context, diags *diag.Diagnostics, clien
...
 		errCode := models.StripQuotes(models.StripSquareBrackets(cont.Search("imdata", "error", "attributes", "code").String()))
-		if errCode != "1" && errCode != "103" && errCode != "107" && errCode != "120" {
+		errText := models.StripQuotes(models.StripSquareBrackets(cont.Search("imdata", "error", "attributes", "text").String()))
+		if errCode == "103" || errCode == "107" || errCode == "202" {
+			tflog.Debug(ctx, errText)
+			return nil
+		} else if (errText == "" && errCode == "403") || errCode == "401" {
+			diags.AddError(
+				"Unable to authenticate. Please check your credentials",
+				fmt.Sprintf("Err: %s. Please report this issue to the provider developers.", err),
```
[the remark this answers is not in the document]
Comment (shrsr) [reply internal/provider/utils.go]: Good call. Changed the error statement

**Review by samiib — APPROVED** [review]
LGTM!

**Review by akinross — APPROVED** [review]
LGTM

**Review by anvitha-jain — APPROVED** [review]
LGTM

**Review by gmicol — APPROVED** [review]
LGTM

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**